src/render/src/device.rs

```diff
@@ -492,6 +492,51 @@ impl<B: Backend> Device<B> {
             .map(Typed::new)
     }
 
+    #[doc(hidden)]
+    pub fn allocate_descriptor_sets(
+        &mut self,
+        sets_bindings: &[&[core::pso::DescriptorSetLayoutBinding]],
```

Comment:
why is this a 2d array?

Reply:
You can allocate multiple sets in a single batch, allowing to share the underlying descriptor pool.